tasks/gcp/StatisticalPhasingTasks.wdl

```diff
@@ -85,11 +87,76 @@ task ShapeIt4 {
   }
 
   output {
-      File phased_vcf =  "~{project_id}_phased.vcf.gz"
+      File region_phased_vcf =  "~{output_filename}"
       File log_file =  "phased.log"
   }
 }
 
+
+
+task LigateRegions {
+  input {
+    Array[File] region_phased_vcfs
+    Array[File] region_phased_vcfs_indices
+    File interval_list
+    String project_id
+
+    String docker_tag = "us.gcr.io/broad-gotc-prod/malariagen/bcftoolspython:1.11"
+    Int preemptible_tries = runTimeSettings.preemptible_tries
+    Int num_cpu = 1
+    RunTimeSettings runTimeSettings
+  }
+
+  Int disk_size = ceil(size(region_phased_vcfs, "GiB")) * 2 + 20
+
+  command {
+    set -e pipefail
+
+    python3 <<CODE
+    from pathlib import Path
+
+    index_files = [ "~{sep='", "' region_phased_vcfs_indices}" ]
+    for f in index_files:
+        Path(f).touch()
+
+    input_files = [ "~{sep='", "' region_phased_vcfs}" ]
+    with open("~{interval_list}") as f:
+        intervals = [i.replace(":", "_").strip("\n") for i in f]
+    ordered_files = []
+
+    for i in intervals:
+        for f in input_files:
+            if i in f:
+                ordered_files.append(f)
+
+    with open("phased_vcf_list.txt", "w") as f:
+        f.write("\n".join(ordered_files))
+    CODE
+
+    bcftools concat \
+        --file-list "phased_vcf_list.txt" \
+        --ligate \
+        --output ~{project_id}_phased.vcf
+
+    bgzip ~{project_id}_phased.vcf
+    tabix ~{project_id}_phased.vcf.gz
+  }
+
+  runtime {
+    docker: docker_tag
+    preemptible: preemptible_tries
+    cpu: num_cpu
+    memory: "15 GiB"
```

Comment:
Should this be a task-level argument?

Reply:
good idea, I have made memory a task level argument for all tasks in this wdl